### PR TITLE
[Edit Price] Add edit budget

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
Disposable Edit Price:
image

Recurring Edit Price:
image

### CODE REVIEW

**Comment by Costa769** [comment]
For both recurring and disposable strategies, if user:

1. Clicks on edit prices
2. Enters deposit/withdraw amount in one or both budget fields
3. No prices are edited, or strategy type changed and no warning messages are displayed

Expected:

- No checkbox is displayed
- The CTA is enabled

Actual:

- Checkbox is displayed
- After checking the checkbox the CTA remains disabled

**Comment by Costa769** [comment]
- [ ] Order type selection

Move order type selection below the token pair on edit price page
image


- [ ] Budget changes section

In case a  user opens a disposable strategy edit prices page or opens edit prices page of recurring or concentrated strategy and changes order type to disposable, the following logic should be implemented
IF inactive order budget != 0
- Show a budget changes section 
- Show a warning inside the budget section (which will force a checkbox to be displayed, which in tern will enable the CTA)
-- Warning text- "
- On CTA click the entire inactive order budget should be withdrawn to the user's wallet

ELSE

Do not show  both the budget changes section and warning 


image

In case user changes the order type from concentrated/recurring to disposable and , display order removed notification inside the Budget changes section

-[ ] Withdraw budget notification

**Comment by Costa769** [comment]
- [ ] Disposable strategy CTA

Go to disposable strategy edit prices page without making  any change to prices/budgets or or order type

Expected:
CTA is disabled
Actual: 
CTA is enabled

- [ ] Warning message text
The warning text inside the Budget changes section should always be "You will withdraw the following funds to your wallet"

File: src/components/strategies/overlapping/OverlappingAction.tsx [hunk 1]
```diff
@@ -20,7 +19,7 @@ interface Props {
   anchor: 'buy' | 'sell';
   action?: BudgetAction;
   setAction: (action: BudgetAction) => void;
-  error: string;
+  error?: string;
```
Comment (tiagofilipenunes) [inline src/components/strategies/overlapping/OverlappingAction.tsx:22]: The error prop is never set outside of the OverlappingAction component

Comment (zavelevsky) [reply src/components/strategies/overlapping/OverlappingAction.tsx]: removed unused prop

**Review by tiagofilipenunes — APPROVED** [review]

**Comment by Costa769** [comment]
- [ ] Replace budget changes section with warning

On edit prices disposable tab page load:

**IF** the inactive order budget !=0 (note that inactive order budget refers to the budget of an order that is being removed due to the order type change or the budget gained from the execution of a disposable order):

- Do not show budget changes section.
- Show a withdraw budget warning instead [mockup]() (This will force a checkbox to appear, which will enable the CTA, as currently implemented).
- Warning text:
- **IF** inactive order is a **sell** order
"You will withdraw[token amount] [token symbol] from the inactive sell order to your wallet"
- **ELSE,IF** inactive order is a **buy** order
"You will withdraw[token amount] [token symbol] from the inactive buy order to your wallet"



- On CTA click, the entire inactive order budget will be withdrawn to the user's wallet.

**ELSE:**

Do not display  a warning text.

**Pull request merged**